app/models/fact_value.rb

```diff
@@ -2,7 +2,7 @@ class FactValue < ActiveRecord::Base
   include Authorizable
   include ScopedSearchExtensions
 
-  belongs_to_host
+  belongs_to :host, {:class_name => "Host::Base", :foreign_key => :host_id}
```

Comment:
Facts are connected to Host::Base. This will allow removing a workaround in discovery that causes n+1 queries on fact pages.